### PR TITLE
magic trailing comma for ExprList

### PR DESCRIPTION
**Pull request by davidszotten**
seems mostly better, still need to think about

`json = {"k": {"k2": {"k3": [1,]}}}` where we differ from black in trailing commas for all the outer dicts

### CODE REVIEW

File: crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__black_test__function_trailing_comma_py.snap [hunk 1]
```diff
@@ -113,23 +113,24 @@ some_module.some_function(
...
--    }
 +def xxxxxxxxxxxxxxxxxxxxxxxxxxxx() -> Set[
 +    "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
 +]:
-+    json = {"k": {"k2": {"k3": [1]}}}
+     json = {
+         "k": {
+             "k2": {
+                 "k3": [
+                     1,
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__black_test__function_trailing_comma_py.snap:123]: Interesting that black doesn't add trailing commas to all outer elements, especially considering that the trailing comma in the array expression isn't JSON.

We can tackle this divergence as its own PR if you like. It probably requires:
* Detecting that the dict (and list) only contains JSON like values
* Add a `Preserve` options to the `comma_separated` builder to only preserve existing trailing commas.

**Review by konstin — APPROVED** [review]

**Review by MichaReiser — APPROVED** [review]

**Comment by davidszotten** [comment]
so many merge conflicts today :(

**Comment by davidszotten** [comment]
hm i can't reproduce this test failure locally

**Comment by konstin** [comment]
Could you try `cargo insta test --all --all-features --delete-unreferenced-snapshots`? That should delete the extra snapshot.

**Comment by davidszotten** [comment]
ah, thanks

i was very confused because my original patch didn't touch that file but i realise now that my patch +the most recent rebase made ruff match black exactly for this file which removes the snap file

**Comment by MichaReiser** [comment]
> so many merge conflicts today :(

I'm sorry 😞

**Comment by davidszotten** [comment]
> > so many merge conflicts today :(
> 
> I'm sorry 😞

no worries, all good

**Pull request merged**